File: productos_app_frontend/src/types/Producto.ts
import { Usuario } from "./Usuario";
export interface Producto {
  id: string;
  nombre: string;
  descripcion: string;
  precio: number;
  estado: boolean;
  usuarioCreacionId: string;
  fechaCreacion: string;
  usuarioModificacionId: string | null;
  fechaModificacion: string | null;
  usuarioCreacion: Usuario | null;
}

File: productos_app_api/Models/Producto.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace productos_app_api.Models;

public class Producto
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public required string Nombre { get; set; }
    public string? Descripcion { get; set; }
    public required decimal Precio { get; set; }
    public required bool Estado { get; set; }
    public required Guid UsuarioCreacionId { get; set; }
    public DateTime FechaCreacion { get; set; } = DateTime.Now;
    public Guid? UsuarioModificacionId { get; set; }
    public DateTime? FechaModificacion { get; set; }
    public  Usuario? UsuarioCreacion { get; set; }
    public  Usuario? UsuarioModificacion { get; set; }
}
